### ISSUE TITLE
Unable to acquire the dpkg frontend lock

### ISSUE THREAD

**Issue by sed-i** [canonical/charmcraft#567]
This only happened to me once but hopefully the attached log would be helpful enough.

A pytest-operator integration test that I ran locally failed (only once so far) with:

```shell
Traceback (most recent call last):
  File "/home/tux/charms/karma-alertmanager-proxy-operator/tests/integration/test_charm_integ.py", line 24, in test_build_and_deploy
    charm_under_test = await ops_test.build_charm(".")
  File "/home/tux/charms/karma-alertmanager-proxy-operator/.tox/integration/lib/python3.9/site-packages/pytest_operator/plugin.py", line 357, in build_charm
    raise RuntimeError(
RuntimeError: Failed to build charm .:
Packing charm 'karma-alertmanager-proxy-k8s_ubuntu-20.04-amd64.charm'...
Failed to install apt-utils.
* Command that failed: 'lxc --project charmcraft exec local:charmcraft-karma-alertmanager-proxy-k8s-56754252-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get install -y apt-utils'
* Command exit code: 100
* Command standard error output: b'E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 653 (apt-get)\nE: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?\n' (full execution logs in '/home/tux/snap/charmcraft/common/charmcraft-log-8jlotfnh')
```
[charmcraft.log](https://github.com/canonical/charmcraft/files/7308622/charmcraft.log)

**Comment by sed-i** [comment]
This also an occasional issue in 1.2.1 as well as 1.3.1.
Here's a log from 1.2.1.
```
Packing charm 'lma-rules-k8s_ubuntu-20.04-amd64.charm'...
charmcraft internal error! BaseConfigurationError: Failed to setup snapd.
* Command that failed: 'lxc --project charmcraft exec local:charmcraft-lma-rules-k8s-1180934-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get install -y snapd'
* Command exit code: 100
* Command standard error output: b'E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 611 (apt-get)\nE: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?\n' (full execution logs in /home/ubuntu/snap/charmcraft/common/charmcraft-log-fjsqzo1m)
```
[cc121.txt](https://github.com/canonical/charmcraft/files/7540067/cc121.txt)

**Comment by facundobatista** [comment]
We will [disable the automatic updates in Craft Providers](https://github.com/canonical/craft-providers/issues/104) for this to stop happening.

**Comment by facundobatista** [comment]
Already solved in #745 where we don't update the APT cache everytime we use the container, so we stop clashing with ad-hoc APT usage.

**Comment by sed-i** [comment]
Happened once more with 2.1.0 (a subsequent `pack` worked fine).

```
2022-10-26 20:40:39.016 Starting charmcraft version 2.1.0
2022-10-26 20:40:39.020 System details: OSPlatform(system='ubuntu', release='22.04', machine='x86_64'); Environment: None
2022-10-26 20:40:39.020 Packing the charm.
2022-10-26 20:40:39.020 Reading '/home/ubuntu/cos-configuration-k8s-operator/metadata.yaml'
2022-10-26 20:40:39.025 Validating metadata format
2022-10-26 20:40:41.335 Executing on host: lxc --project default profile show local:default
2022-10-26 20:40:48.777 Building for 'bases[0]' as host matches 'build-on[0]'.
2022-10-26 20:40:48.778 Building for 'bases[0][0]'.
2022-10-26 20:40:48.778 Launching environment to pack for base name='ubuntu' channel='20.04' architectures=['amd64'] (may take a while the first time but it's reusable)
2022-10-26 20:40:48.783 Executing on host: lxc remote list --format=yaml
2022-10-26 20:40:49.303 Remote 'craft-com.ubuntu.cloud-buildd' already exists.
2022-10-26 20:40:49.304 Using hostname 'charmcraft-cos-configuration-k8s-523758-0-0-amd64'
2022-10-26 20:40:49.306 Executing on host: lxc project list local: --format=yaml
2022-10-26 20:40:49.577 Set LXD instance name to 'charmcraft-cos-configuration-k8s-523758-0-0-amd64'
2022-10-26 20:40:49.578 Executing on host: lxc --project charmcraft list local: --format=yaml
2022-10-26 20:40:49.824 Executing on host: lxc --project charmcraft image list local: --format=yaml
2022-10-26 20:40:49.948 Using compatible snapshot 'snapshot-craft-com.ubuntu.cloud-buildd-20.04-charmcraft-buildd-base-v0.0'.
2022-10-26 20:40:49.949 Executing on host: lxc --project charmcraft info local:
2022-10-26 20:40:50.120 Executing on host: lxc --project charmcraft launch local:snapshot-craft-com.ubuntu.cloud-buildd-20.04-charmcraft-buildd-base-v0.0 local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 --config 'raw.idmap=both 1000 0' --config security.syscalls.intercept.mknod=true
2022-10-26 20:41:09.384 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 cat /etc/os-release
2022-10-26 20:41:10.967 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 test -f /etc/craft-instance.conf
2022-10-26 20:41:11.709 Executing on host: lxc --project charmcraft file pull local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/craft-instance.conf /home/ubuntu/tmpahhbtgym.tmp-craft/tmpg50mq18b
2022-10-26 20:41:12.797 Instance is compatible with compatibility tag 'charmcraft-buildd-base-v0.0'
2022-10-26 20:41:12.806 Executing on host: lxc --project charmcraft file push /home/ubuntu/snap/charmcraft/common/tmpoqqxmeie local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/apt/apt.conf.d/20auto-upgrades --mode=0644
2022-10-26 20:41:13.871 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 chown root:root /etc/apt/apt.conf.d/20auto-upgrades
2022-10-26 20:41:15.106 Executing on host: lxc --project charmcraft file push /home/ubuntu/snap/charmcraft/common/tmpa7y7j899 local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/environment --mode=0644
2022-10-26 20:41:15.872 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 chown root:root /etc/environment
2022-10-26 20:41:16.734 Waiting for environment to be ready...
2022-10-26 20:41:16.734 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:17.599 systemctl is-system-running status: starting
2022-10-26 20:41:17.850 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:18.787 systemctl is-system-running status: starting
2022-10-26 20:41:19.038 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:19.582 systemctl is-system-running status: starting
2022-10-26 20:41:19.833 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:20.854 systemctl is-system-running status: starting
2022-10-26 20:41:21.105 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:22.633 systemctl is-system-running status: starting
2022-10-26 20:41:22.883 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:24.453 systemctl is-system-running status: starting
2022-10-26 20:41:24.704 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:26.170 systemctl is-system-running status: starting
2022-10-26 20:41:26.421 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:27.288 systemctl is-system-running status: starting
2022-10-26 20:41:27.541 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:28.172 systemctl is-system-running status: starting
2022-10-26 20:41:28.423 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:28.796 systemctl is-system-running status: starting
2022-10-26 20:41:29.046 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:29.869 systemctl is-system-running status: starting
2022-10-26 20:41:30.120 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:31.061 systemctl is-system-running status: starting
2022-10-26 20:41:31.311 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:32.093 systemctl is-system-running status: starting
2022-10-26 20:41:32.344 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:33.324 systemctl is-system-running status: starting
2022-10-26 20:41:33.575 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:34.212 systemctl is-system-running status: starting
2022-10-26 20:41:34.463 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl is-system-running
2022-10-26 20:41:35.228 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 test -f /etc/craft-instance.conf
2022-10-26 20:41:35.391 Executing on host: lxc --project charmcraft file pull local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/craft-instance.conf /home/ubuntu/tmp3p9_l1cc.tmp-craft/tmptqd0_7dt
2022-10-26 20:41:35.626 Executing on host: lxc --project charmcraft file push /home/ubuntu/snap/charmcraft/common/tmpjrnpfp_w local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/craft-instance.conf --mode=0644
2022-10-26 20:41:35.859 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 chown root:root /etc/craft-instance.conf
2022-10-26 20:41:36.027 Executing on host: lxc --project charmcraft file push /home/ubuntu/snap/charmcraft/common/tmpnef9rben local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/hostname --mode=0644
2022-10-26 20:41:36.192 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 chown root:root /etc/hostname
2022-10-26 20:41:36.493 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 hostname -F /etc/hostname
2022-10-26 20:41:36.831 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 ln -sf /run/systemd/resolve/resolv.conf /etc/resolv.conf
2022-10-26 20:41:37.227 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl enable systemd-resolved
2022-10-26 20:41:40.088 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl restart systemd-resolved
2022-10-26 20:41:46.788 Executing on host: lxc --project charmcraft file push /home/ubuntu/snap/charmcraft/common/tmp1h1j5pfs local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/systemd/network/10-eth0.network --mode=0644
2022-10-26 20:41:47.257 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 chown root:root /etc/systemd/network/10-eth0.network
2022-10-26 20:41:47.763 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl enable systemd-networkd
2022-10-26 20:41:48.675 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl restart systemd-networkd
2022-10-26 20:41:50.839 Waiting for networking to be ready...
2022-10-26 20:41:50.839 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 getent hosts snapcraft.io
2022-10-26 20:41:51.296 Executing on host: lxc --project charmcraft file push /home/ubuntu/snap/charmcraft/common/tmp2s4rgtra local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/apt/apt.conf.d/00no-recommends --mode=0644
2022-10-26 20:41:51.626 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 chown root:root /etc/apt/apt.conf.d/00no-recommends
2022-10-26 20:41:51.814 Executing on host: lxc --project charmcraft file push /home/ubuntu/snap/charmcraft/common/tmp54p_p_j5 local:charmcraft-cos-configuration-k8s-523758-0-0-amd64/etc/apt/apt.conf.d/00update-errors --mode=0644
2022-10-26 20:41:51.957 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 chown root:root /etc/apt/apt.conf.d/00update-errors
2022-10-26 20:41:52.133 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get update
2022-10-26 20:41:58.681 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get install -y apt-utils curl
2022-10-26 20:42:03.691 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get install -y fuse udev
2022-10-26 20:42:05.146 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl enable systemd-udevd
2022-10-26 20:42:05.526 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 systemctl start systemd-udevd
2022-10-26 20:42:05.918 Executing in container: lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get install -y snapd
2022-10-26 20:42:06.524 Failed to setup snapd.
* Command that failed: 'lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get install -y snapd'
* Command exit code: 100
* Command standard error output: b'E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 858 (apt-get)\nE: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?\n'
2022-10-26 20:42:06.539 Traceback (most recent call last):
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/lib/craft_providers/bases/buildd.py", line 730, in _setup_snapd
2022-10-26 20:42:06.539     executor.execute_run(
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/lib/craft_providers/lxd/lxd_instance.py", line 289, in execute_run
2022-10-26 20:42:06.539     return self.lxc.exec(
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/lib/craft_providers/lxd/lxc.py", line 329, in exec
2022-10-26 20:42:06.539     return runner(final_cmd, **kwargs)  # pylint: disable=subprocess-run-check
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/usr/lib/python3.8/subprocess.py", line 516, in run
2022-10-26 20:42:06.539     raise CalledProcessError(retcode, process.args,
2022-10-26 20:42:06.539 subprocess.CalledProcessError: Command '['lxc', '--project', 'charmcraft', 'exec', 'local:charmcraft-cos-configuration-k8s-523758-0-0-amd64', '--', 'env', 'PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin', 'CHARMCRAFT_MANAGED_MODE=1', 'apt-get', 'install', '-y', 'snapd']' returned non-zero exit status 100.
2022-10-26 20:42:06.539 
2022-10-26 20:42:06.539 The above exception was the direct cause of the following exception:
2022-10-26 20:42:06.539 Traceback (most recent call last):
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/lib/charmcraft/providers/_lxd.py", line 151, in launched_environment
2022-10-26 20:42:06.539     instance = lxd.launch(
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/lib/craft_providers/lxd/launcher.py", line 212, in launch
2022-10-26 20:42:06.539     base_configuration.setup(executor=instance)
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/lib/charmcraft/providers/_buildd.py", line 110, in setup
2022-10-26 20:42:06.539     super().setup(executor=executor, retry_wait=retry_wait, timeout=timeout)
2022-10-26 20:42:06.539   File "/snap/charmcraft/1083/lib/craft_providers/bases/buildd.py", line 352, in setup
2022-10-26 20:42:06.540     self._setup_snapd(executor=executor, deadline=deadline)
2022-10-26 20:42:06.540   File "/snap/charmcraft/1083/lib/craft_providers/bases/buildd.py", line 759, in _setup_snapd
2022-10-26 20:42:06.540     raise BaseConfigurationError(
2022-10-26 20:42:06.540 craft_providers.bases.errors.BaseConfigurationError: Failed to setup snapd.
2022-10-26 20:42:06.540 * Command that failed: 'lxc --project charmcraft exec local:charmcraft-cos-configuration-k8s-523758-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 apt-get install -y snapd'
2022-10-26 20:42:06.540 * Command exit code: 100
2022-10-26 20:42:06.540 * Command standard error output: b'E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 858 (apt-get)\nE: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?\n'
2022-10-26 20:42:06.540 Full execution log: '/home/ubuntu/snap/charmcraft/common/cache/charmcraft/log/charmcraft-20221026-204039.014241.log'
```
